hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -46,286 +79,392 @@ def find_target(target_name = ''):
     used_pins = []
     used_pins_friendly = []
     invalid_items = []
 
-    for key, val in pinName_dict.items():
+    for key, val in pin_name_dict.items():
         if re.match("^ARDUINO_UNO_[AD]\d+$", key):
-            if val == 'NC':
+            if val == "NC":
                 continue
             # resolve to literal
             realval = val
             depth = 0
-            while not re.match("(0x[0-9a-fA-F]+|[1-9][0-9]*|0[1-7][0-7]+|0b[01]+)[uUlL]{0,2}", realval):
+            while not re.match(
+                "(0x[0-9a-fA-F]+|[1-9][0-9]*|0[1-7][0-7]+|0b[01]+)[uUlL]{0,2}",
+                realval,
+            ):
                 try:
-                    realval = pinName_dict[realval]
+                    realval = pin_name_dict[realval]
                     depth += 1
-                except:
+                except KeyError:
                     break
                 if depth > 10:
                     break
 
             if realval in used_pins:
-                message = "already assigned to " + used_pins_friendly[used_pins.index(realval)]
-                invalid_items.append({'key': key, 'val': val, 'message': message})
+                message = (
+                    "already assigned to "
+                    + used_pins_friendly[used_pins.index(realval)]
+                )
+                invalid_items.append(
+                    {"key": key, "val": val, "message": message}
+                )
                 continue
 
             used_pins.append(realval)
-            used_pins_friendly.append(key + ' = ' + val)
+            used_pins_friendly.append(key + " = " + val)
     return invalid_items
 
-def arduino_existence_check(pinName_dict):
+
+def arduino_existence_check(pin_name_dict):
     required_pins = [
-        'ARDUINO_UNO_A0',
-        'ARDUINO_UNO_A1',
-        'ARDUINO_UNO_A2',
-        'ARDUINO_UNO_A3',
-        'ARDUINO_UNO_A4',
-        'ARDUINO_UNO_A5',
-        'ARDUINO_UNO_D0',
-        'ARDUINO_UNO_D1',
-        'ARDUINO_UNO_D2',
-        'ARDUINO_UNO_D3',
-        'ARDUINO_UNO_D4',
-        'ARDUINO_UNO_D5',
-        'ARDUINO_UNO_D6',
-        'ARDUINO_UNO_D7',
-        'ARDUINO_UNO_D8',
-        'ARDUINO_UNO_D9',
-        'ARDUINO_UNO_D10',
-        'ARDUINO_UNO_D11',
-        'ARDUINO_UNO_D12',
-        'ARDUINO_UNO_D13',
-        'ARDUINO_UNO_D14',
-        'ARDUINO_UNO_D15'
+        "ARDUINO_UNO_A0",
+        "ARDUINO_UNO_A1",
+        "ARDUINO_UNO_A2",
+        "ARDUINO_UNO_A3",
+        "ARDUINO_UNO_A4",
+        "ARDUINO_UNO_A5",
+        "ARDUINO_UNO_D0",
+        "ARDUINO_UNO_D1",
+        "ARDUINO_UNO_D2",
+        "ARDUINO_UNO_D3",
+        "ARDUINO_UNO_D4",
+        "ARDUINO_UNO_D5",
+        "ARDUINO_UNO_D6",
+        "ARDUINO_UNO_D7",
+        "ARDUINO_UNO_D8",
+        "ARDUINO_UNO_D9",
+        "ARDUINO_UNO_D10",
+        "ARDUINO_UNO_D11",
+        "ARDUINO_UNO_D12",
+        "ARDUINO_UNO_D13",
+        "ARDUINO_UNO_D14",
+        "ARDUINO_UNO_D15",
```

Comment:
This is not technically wrong, but a slightly shorter way to write this would be:

```python
ard_base = 
required_pins = [f"ARDUINO_UNO_A{i}" for i in range(6)] + [f"ARDUINO_UNO_D{i} for i in range(16)]
```

I am assuming here that the minimum supported python version is 3.6. If it is 3.5, then you'll have to suffer with `str.format`.